ui/src/features/components/Report/compareReports.js

```diff
@@ -85,6 +89,46 @@ class CompareReports extends React.Component {
         this.setState({filteredKeys: {...newFilteredKeys}});
     };
 
+    onClickExportCSV = () => {
+        const selectedReports = this.props.selectedReportsAsArray;
+        const reportsList = this.state.reportsList;
+        let request_string = "";
+        let reportIdsAsCSV = "";
+        let testIdsAsCSV = "";
+        for (let index in selectedReports){
+            reportIdsAsCSV+=selectedReports[index]["reportId"];
+            testIdsAsCSV+=selectedReports[index]["testId"];
+            if (index < selectedReports.length -1){
+                reportIdsAsCSV+=",";
+                testIdsAsCSV+=",";
+            }
+        }
+        request_string = "reportIds="+reportIdsAsCSV+"&testIds="+testIdsAsCSV;
+        axios.get(`${process.env.PREDATOR_URL}/tests/reports/compare/export/csv?`+request_string)
```

Comment:
@enudler  what about the case that the data is too big? It will take time for the server to generate the CSV no?
in this case, we can make it like an async request -> The UI will ask to generate CSV, and will get the id for that, then the UI will make checks if the CSV is ready for download.  

if going to keep it simple and saved the async way for the next phase:
you don't need to use Axios for this. you can just open the link in the browser by using window.open.

in general, for API requests we using redux with Saga.

Reply:
@manorlh we don't save the csv anywhere, we don't create any resource
so I think it's ok